### PR TITLE
JWSDK (v3 only) VUDRM Widevine online integration no sdk

### PR DESCRIPTION
**Pull request by macmagician**
JWSDK v3 (only) VUDRM Widevine online integration rebase
Created separate module to preserve original source
Amended required configs to throw syntax error and marked as TODO

### CODE REVIEW

File: VUDRMWidevineDemo/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            android:theme="@style/Theme.AppCompat.Light">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+
+        <!--TODO Enter your license key -->
+        <meta-data
```
Comment (GMSyrimis) [inline VUDRMWidevineDemo/app/src/main/AndroidManifest.xml:26]: License keys are now set programmatically.

File: VUDRMWidevineDemo/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,41 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 29
+
+    defaultConfig {
+        applicationId "com.jwplayer.demo.recyclerview"
```
Comment (GMSyrimis) [inline VUDRMWidevineDemo/app/build.gradle]: May want to change this to match your demo

File: VUDRMWidevineDemo/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.jwplayer.demo.recyclerview">
```
Comment (GMSyrimis) [inline VUDRMWidevineDemo/app/src/main/AndroidManifest.xml]: May want to change this to match your demo

**Review by paulmandal — APPROVED** [review]
Looks good to me!

File: VUDRMWidevineDemo/app/src/main/java/com/jwplayer/demo/recyclerview/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+			// If a player was playing, then it was previously set as the active player, pause() and remove listeners
+			if (player.getState() == PlayerState.PLAYING && !player.equals(mActivePlayer)) {
+				player.pause();
+				mKeepScreenOnHandler.removeListeners(player);
+			}
+		}
+	}
+
+
+	public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
```
Comment (GMSyrimis) [inline VUDRMWidevineDemo/app/src/main/java/com/jwplayer/demo/recyclerview/MainActivity.java]: Any particular reason you chose to include the RecyclerView in your demo? It adds complexity to the demo distracting the developer from required code and optional code.

**Review by GMSyrimis — CHANGES_REQUESTED** [review]

**Review by GMSyrimis — APPROVED** [review]

**Pull request merged**